### PR TITLE
Adds some store tips

### PR DESCRIPTION
**Pull request by tildedave**

### CODE REVIEW

**Comment by tildedave** [comment]
discuss

File: 11-flux.md [hunk 1]
```diff
@@ -33,5 +41,63 @@ module.exports = SessionActions;
...
+
+ With this separation of data, it's really easy to handle "live update" cases.  The page can connect to a websocket and wait for push notifications in the background of updated campaign data.  On a socket notification, the store emits change and the view is rerendered with the new data.  (XHR polling can also be used to accomplish the same thing.)
+
+Stores similarly make "faking data" before a webservice call has completed really straightforward.  When the user clicks 'send' to create a comment, a fake comment is created in the store, and the store signals a change has occurred.  The view rerenders, showing the new comment.  After the comment has been successfully persisted, the store is updated to replace the fake comment's data with the real comments data, and the store emits change again, updating the new comment with its real backing data.  This level of responsiveness is expected by most customers in their interaction with the social features on a website.
+
+### Best Practices
+
+The best implementation of a store:
+* corresponds to a `GET` endpoint from some upstream service (API)
+* stores data is completely parameterized by the `GET` arguments to that endpoint
```
Comment (rtymchyk) [inline 11-flux.md]: store's

File: 11-flux.md [hunk 2]
```diff
@@ -33,5 +41,63 @@ module.exports = SessionActions;
...
+
+ With this separation of data, it's really easy to handle "live update" cases.  The page can connect to a websocket and wait for push notifications in the background of updated campaign data.  On a socket notification, the store emits change and the view is rerendered with the new data.  (XHR polling can also be used to accomplish the same thing.)
+
+Stores similarly make "faking data" before a webservice call has completed really straightforward.  When the user clicks 'send' to create a comment, a fake comment is created in the store, and the store signals a change has occurred.  The view rerenders, showing the new comment.  After the comment has been successfully persisted, the store is updated to replace the fake comment's data with the real comments data, and the store emits change again, updating the new comment with its real backing data.  This level of responsiveness is expected by most customers in their interaction with the social features on a website.
```
Comment (rtymchyk) [inline 11-flux.md]: I like this.

File: 11-flux.md [hunk 3]
```diff
@@ -33,5 +41,63 @@ module.exports = SessionActions;
...
+
+### Best Practices
+
+The best implementation of a store:
+* corresponds to a `GET` endpoint from some upstream service (API)
+* stores data is completely parameterized by the `GET` arguments to that endpoint
+
+A store backing the Tilt page - upstream API route `/tilts/:id`, would contain data inside of it parameterized by Tilt `id`.  This allows for storing data for multiple tilts in memory at once, which allows seamless transitions between different tilts without a page refresh.  As an example, we could preload data for a tilt when a user mouses over a notification in their notification center - on click, the page could be loaded automatically from memory, avoiding a loading screen.
```
Comment (rtymchyk) [inline 11-flux.md]: Maybe put this under another point above and use this as examples for that.

File: 11-flux.md [hunk 4]
```diff
@@ -33,5 +41,63 @@ module.exports = SessionActions;
...
+
+This can lead to somewhat silly conclusions: "`/users/:id/notifications` should be parameterized by user".  At first thought this is a bad conclusion, because only one user is ever logged in at once.  However, a store that is parameterized by user id would allow quickly switching between "active accounts", or displaying data from multiple accounts on the screen at once.  Building a store like this for present product requirements may be overdesign (and so not worth it) but it is good to think through these edge cases and understand the tradeoff that is being made ;)
+
+### Try To Avoid
+
+Try to avoid the store pattern for:
+* data that better represents view state, e.g. errors temporarily displayed, submission in progress, or successfully finished
+* data that will never change (this is better owned and passed down as props from some parent component)
+
```
Comment (rtymchyk) [inline 11-flux.md]: Add a point to avoid complicated logic on the data inside a store and refer to model use for that.

**Comment by rtymchyk** [comment]
This is really helpful @tildedave, thanks!

File: 11-flux.md (hunk 4 above)
Comment (tildedave) [inline 11-flux.md]: good call, updated.  hopefully we should have some better examples with redux

**Comment by tildedave** [comment]
This is mostly still accurate but now that we're on Redux the questions are a little different.

**Comment by amackera** [comment]
Great info here!

**Pull request merged**